tailcall-typedefs/src/gen_gql_schema.rs

```diff
@@ -1,723 +1,181 @@
-
-        write!(f, " on ")?;
-        first.to_graphql(f)?;
-
-        for entry in iter {
-            write!(f, " | ")?;
-            entry.to_graphql(f)?;
-        }
-
-        write!(f, "\n\n")
-    }
-}
-
-struct LineBreaker<'a> {
-    string: &'a str,
-    break_at: usize,
-    index: usize,
-}
-
-impl<'a> LineBreaker<'a> {
-    fn new(string: &'a str, break_at: usize) -> Self {
-        LineBreaker { string, break_at, index: 0 }
-    }
-}
-
-impl<'a> Iterator for LineBreaker<'a> {
-    type Item = &'a str;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        if self.index >= self.string.len() {
-            return None;
-        }
-
-        let end_index = self
-            .string
-            .chars()
-            .skip(self.index + self.break_at)
-            .enumerate()
-            .find(|(_, ch)| ch.is_whitespace())
-            .map(|(index, _)| self.index + self.break_at + index + 1)
-            .unwrap_or(self.string.len());
-
-        let start_index = self.index;
-        self.index = end_index;
-
-        Some(&self.string[start_index..end_index])
-    }
-}
-
-struct IndentedWriter<W: Write> {
-    writer: W,
-    indentation: usize,
-    line_broke: bool,
-}
-
-impl<W: Write> IndentedWriter<W> {
-    fn new(writer: W) -> Self {
-        IndentedWriter { writer, indentation: 0, line_broke: false }
-    }
-
-    fn indent(&mut self) {
-        self.indentation += 2;
-    }
-
-    fn unindent(&mut self) {
-        self.indentation -= 2;
-    }
-}
-
-impl<W: std::io::Write> Write for IndentedWriter<W> {
-    #[allow(clippy::same_item_push)]
-    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
-        let mut new_buf = vec![];
-        let mut extra = 0;
-
-        for ch in buf {
-            if self.line_broke && self.indentation > 0 {
-                extra += self.indentation;
-                for _ in 0..self.indentation {
-                    new_buf.push(b' ');
-                }
-            }
-            self.line_broke = false;
-
-            new_buf.push(*ch);
-            if ch == &b'\n' {
-                self.line_broke = true;
-            }
-        }
-
-        self.writer.write(&new_buf).map(|a| a - extra)
-    }
-
-    fn flush(&mut self) -> std::io::Result<()> {
-        self.writer.flush()
-    }
-}
-
-#[derive(Debug)]
-enum ExtraTypes {
-    Schema,
-    ObjectValidation(ObjectValidation),
-}
+use tailcall::core::scalar::{
+    Bytes, Date, Email, Empty, Int128, Int16, Int32, Int64, Int8, PhoneNumber, UInt128, UInt16,
+    UInt32, UInt64, UInt8, Url, JSON,
+};
+use tailcall::core::FileIO;
+use tailcall_typedefs_common::directive_definition::DirectiveDefinition;
+use tailcall_typedefs_common::input_definition::InputDefinition;
+use tailcall_typedefs_common::scalar_definition::ScalarDefinition;
+use tailcall_typedefs_common::ServiceDocumentBuilder;
+
+static GRAPHQL_SCHEMA_FILE: &str = "generated/.tailcallrc";
+
+pub async fn update_gql(file_io: Arc<dyn FileIO>) -> Result<()> {
```

Comment:
The file operation doesn't need to be here. We should generate and return a ServiceDocument here.